src/extremeweatherbench/case.py

```diff
@@ -30,165 +32,99 @@ class IndividualCase:
-            metrics.MaximumMAE,
-        ]
-    )
-    data_vars: List[str] = dataclasses.field(
-        default_factory=lambda: ["surface_air_temperature"]
-    )
+class IndividualCaseCollection:
+    """A collection of IndividualCases."""
 
-    def __post_init__(self):
-        self.data_vars = ["surface_air_temperature"]
+    cases: list[IndividualCase]
 
 
 @dataclasses.dataclass
-class IndividualFreezeCase(IndividualCase):
-    """Container for metadata defining a single or individual case of a freeze event.
+class CaseOperator:
+    """A class which stores the graph to process an individual case.
 
-    An IndividualFreezeCase is a subclass of IndividualCase that is designed to
-    provide additional metadata specific to freeze events.
+    This class is used to store the graph to process an individual case. The purpose of
+    this class is to be a one-stop-shop for the evaluation of a single case. Multiple
+    CaseOperators can be run in parallel to evaluate multiple cases, or run through the
+    ExtremeWeatherBench.run() method to evaluate all cases in an evaluation in serial.
 
     Attributes:
-        freeze_type: str: A string representing the type of freeze event.
+        case: IndividualCase metadata
+        metric: A metric that is intended to be evaluated for the case
+        target_config: A TargetConfig object
+        forecast_config: A ForecastConfig object
     """
 
-    metrics_list: List[Type[metrics.Metric]] = dataclasses.field(
-        default_factory=lambda: [metrics.RegionalRMSE]
-    )
-    data_vars: List[str] = dataclasses.field(
-        default_factory=lambda: [
-            "surface_air_temperature",
-            "surface_eastward_wind",
-            "surface_northward_wind",
-        ]
-    )
+    case: IndividualCase
+    metric: "metrics.BaseMetric"
+    target: "inputs.TargetBase"
+    forecast: "inputs.ForecastBase"
 
-    def __post_init__(self):
-        self.data_vars = [
-            "surface_air_temperature",
-            "surface_eastward_wind",
-            "surface_northward_wind",
-        ]
 
+def build_case_operators(
+    cases_dict: dict[str, list],
+    metric_evaluation_objects: list["inputs.EvaluationObject"],
+) -> list[CaseOperator]:
+    """Build a CaseOperator from the case metadata and metric evaluation objects.
 
-# maps the case event type to the corresponding dataclass
-# additional event types need to be added here and
-# CASE_EVENT_TYPE_MATCHER, which maps the metadata case event type
-# to the corresponding case dataclass.
-class CaseEventType(StrEnum):
-    """Enum class for the different types of extreme weather events."""
+    Args:
+        cases_dict: The case metadata to use for the case operators.
+        metric_evaluation_objects: The metric evaluation objects to use for the case operators.
 
-    HEAT_WAVE = "heat_wave"
-    FREEZE = "freeze"
+    Returns:
+        A list of CaseOperator objects.
+    """
+    case_metadata_collection = dacite.from_dict(
```

Comment:
Sorry for being vague but this just doesn't feel right. Generally, parsing of config(s) should happen once, at the top-level entry point to the system.